wireguard-go-rust-wrapper/wireguard-go/platform_specific_windows.go

```diff
@@ -24,10 +57,21 @@ func PlatformSpecific_CreateTun(ifname string) (*tun.NativeTun, error) {
 	networkProfileGuidString := "{" + networkProfileUuid.String() + "}"
 	networkProfileGuid, _ := windows.GUIDFromString(networkProfileGuidString)
 
+	exists := CheckIfAdapterExists(networkProfileGuidString)
+
 	infof("Creating wintun interface %s with GUID %s", ifname, networkProfileGuidString)
 	wintun, err := tun.CreateTUNWithRequestedGUID(ifname, &networkProfileGuid, 0)
 	if err != nil || wintun == nil {
-		return nil, err
+		// If TUN with same GUID exists, try to create TUN one more time,
```

Comment:
It was mentioned that if you create and it fails, Windows does some cleanup and the second time it might succeed. Could it be a bit simpler to just try two times instead?

Other than that we have `exists` already, maybe we can destroy it beforehand?

Reply:
Wintun does some sort of orphaned interface cleanup, but since we do not use wintun directly, instead we use wireguard-go, which doesn't expose that cleanup function, I just let wintun itself to handle this thing.